snakefmt/formatter.py

```diff
@@ -128,7 +134,13 @@ def flush_buffer(
 
         # Re-add newline removed by black for proper parsing of comments
         if self.buffer.endswith("\n\n"):
-            formatted += "\n"
+            if comment_start(self.buffer.rstrip().splitlines()[-1]):
```

Comment:
Would this fail if there was a space *before* the `#` in a comment? i.e. would `strip` be more appropriate?

Reply:
Great spot. Because we're splitting lines here, `strip` wont be enough: i've added `string.lstrip().startswith("#)` is `cmment_start` function to handle this

Reply:
The buffer itself i think should stay as rstrip: if there are only comments and space below, sticks `include` (or whatever) together, if space and comments below, keeps them distinct